packages/devtools_app/lib/src/logging/logging_controller.dart

```diff
@@ -635,6 +609,58 @@ class LoggingController {
       ),
     );
   }
+
+  @override
+  List<LogData> matchesForSearch(String search) {
+    if (search == null || search.isEmpty) return [];
+    final matches = <LogData>[];
+    final caseInsensitiveSearch = search.toLowerCase();
+
+    final currentLogs = filteredData.value;
+    for (final log in currentLogs) {
+      if ((log.summary != null &&
+              log.summary.toLowerCase().contains(caseInsensitiveSearch)) ||
+          (log.details != null &&
+              log.details.toLowerCase().contains(caseInsensitiveSearch))) {
+        matches.add(log);
+      }
+    }
+    return matches;
+  }
+
+  @override
+  void filterData(QueryFilter filter) {
+    if (filter == null) {
+      filteredData.value = List.from(data);
+    } else {
+      filteredData.value = data.where((log) {
+        for (final arg in filter.filterArguments) {
```

Comment:
maybe process the filterArugments into a `Map<filterId,  List<Filter>>` so this logic is a little cleaner and slightly more efficient for the case of a really long log.